### ISSUE TITLE
Make --forbid-only (but not --forbid-pending) default

### ISSUE THREAD

**Issue by ScottFreeCode** [mochajs/mocha#3030]
Eventually let's make `--forbid-only` the default and use `--no-forbid-only` to disable it. Lots of CI will get safer for free!

(Pending, on the other hand, isn't so dangerous to begin with and has more long-term use-cases. It'll be inconsistent, but let's leave pending as-is.)

**Comment by boneskull** [comment]
Actually, this is fine if we make `--forbid-only` the default if the `CI` environment var is present.

**Comment by ParkSB** [comment]
I agree with making `--forbid-only` the default on CI environment. Can I take this issue?

**Comment by outsideris** [comment]
@ParkSB Yes, you can.

**Comment by JoshuaKGoldberg** [comment]
I'm in favor of this as a changed default for CI environments. It's better to have safety be opt-in IMO. This is what Vitest does and it's quite useful.

**Comment by JoshuaKGoldberg** [comment]
Forbidding `.only` in CI is a good safe practice that other test frameworks do in CI (e.g. https://vitest.dev/config/#allowonly). Marking as accepting PRs for Mocha.
